### PR TITLE
Upgrade cool.io to 1.2.x

### PR DESCRIPTION
**Pull request by joker1007**
we cannot build cool.io-1.1.1 on ruby-2.2.0.
I fixed cool.io version dependency, use `cool.io ~> 1.2.4`.
And I fixes broken specs, old spec_helper, old Rakefile.

### CODE REVIEW

File: spec/unit/client_spec.rb [hunk 1]
```diff
@@ -34,23 +35,26 @@
...
+	    	result.should equal 3
 		error.should be_nil
 	    end
+	    req.join
 	 end
 
 	 it 'should return nil values when you call notify' do
 	    @client.notify(:hello).should be_nil
 	    @client.notify(:sum,1,2).should be_nil
+      sleep 1
```
Comment (sonots) [inline spec/unit/client_spec.rb]: indent

**Comment by sonots** [comment]
:+1:

File: spec/unit/client_spec.rb (hunk 1 above)
Comment (joker1007) [inline spec/unit/client_spec.rb]: Thanks!
I want to rewrite all to space indent style :sweat_smile:

**Comment by repeatedly** [comment]
:+1:

**Comment by suma** [comment]
:+1:  Thank you for your PR.
I got error [like this](https://gist.github.com/suma/0b28f5465c4b5adb6cde) when `rake rspec` executed with Ruby 2.2.0 and 2.1.5. Do you have any idea?

[note] Indentation is used spaces and tabs. Original code was written with tabs, so we(maintainer) should refine only newer specs.

**Comment by joker1007** [comment]
@suma 
I think that this is because port choice is not safely.
I fixed way of launching server, test is more stable.

**Comment by suma** [comment]
@joker1007 Thanks a lot!
I released v0.5.3.

**Pull request merged**